Lib/python/std_filesystem.i

```diff
@@ -0,0 +1,85 @@
+/*
+  filesystem::path
+*/
+%{
+#if __cplusplus >= 201703L
+#include <filesystem>
+#include <Python.h>
+#endif // __cplusplus >= 201703L
+%}
+
+%fragment("common", "header") {
+    PyObject * importPathCls() {
+        PyObject * module = PyImport_ImportModule("pathlib");
+        PyObject * cls = PyObject_GetAttrString(module, "Path");
+        Py_DECREF(module);
+        return cls;
+    }
+
+    bool isPathInstance(PyObject * obj) {
+        PyObject * cls = importPathCls();
+        bool is_instance =  PyObject_IsInstance(obj, cls);
+        Py_DECREF(cls);
+        return is_instance;
+    }
+
+    const char * pathToStr(PyObject * obj) {
+        PyObject * str_obj = PyObject_Str(obj);     // New reference
+        const char * s = PyUnicode_AsUTF8(str_obj); // This stores the UTF-8 representation buffer within str_obj
+        // Py_DECREF(str_obj);                      // So we don't decref here
+        return s;
+    }
+}
+
+%typemap(in, fragment="common") std::filesystem::path {
+    if (PyUnicode_Check($input)) {
+        const char* s = PyUnicode_AsUTF8($input);
+        $1 = std::filesystem::path(s);
+    }
+    else if (isPathInstance($input)) {
+        const char * s = pathToStr($input);
+        $1 = std::filesystem::path(s);
+    }
```

Comment:
https://github.com/swig/swig/pull/1999#discussion_r902562620

> Are you sure that this call will decode UTF-8 on Windows? If I read the specs correct, this is only done for `char8_t`, so only for unsigned char.
> 
> Should there be checks in the test-suite which checks for unicode edge cases instead of just foo?